core/src/main/java/org/apache/calcite/plan/volcano/RuleQueue.java

```diff
@@ -196,7 +196,7 @@ VolcanoRuleMatch popMatch(VolcanoPlannerPhase phase) {
     phaseMatchList.matchMap.remove(
         planner.getSubset(match.rels[0]), match);
 
-    LOGGER.debug("Pop match: {}", match);
+    LOGGER.trace("Pop match: {}", match);
     return match;
```

Comment:
Any strong reason why switch from `DEBUG` to `TRACE` ?

Reply:
The DEBUG messages would be too verbose to be useful if we keep this at DEBUG.

Reply:
Maybe reasonable, but it is another topic, should be resolved in another issue and it is better if there is a discussion about this switch.

Reply:
Why not just discuss here? Right now there are just too many logs at DEBUG level, including all the rule firing and transformations. I try to separate DEBUG and TRACE so we could get the important ones when enable DEBUG, things like the best plan, rules attempted, etc. Without the log level change, this PR is not quite useful.

Reply:
> Without the log level change, this PR is not quite useful.

@xndai , this is exactly why I suggest that we should use **categories** rather than `log levels` to configure planner tracing.

What you add looks more like a standalone debugging feature rather than a yet another logging level.
We should move heavyweight logging to their own categories otherwise we never converge on what is debug and what is trace.

Reply:
Now I use a separate logger for RuleAttemptsInfo. Please check.